Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -479,6 +440,42 @@ DocumentKey Serializer::DecodeKey(Reader* reader,
   return DocumentKey{std::move(local_path)};
 }
 
+pb_bytes_array_t* Serializer::EncodeQueryPath(const ResourcePath& path) const {
```

Comment:
This is a prerequisite to access controlling `ResourcePath::FromStringView` in c66dc31301ca369f77d16747f883e02cdcd4a67a.

Reply:
Hmm, I don't think it would make any difference for friendship if these were `static` member functions instead of `const` member functions, but perhaps I'm missing something. Anyway, feel free to keep as is.

Reply:
Previously these functions were `static` in the sense of being members of the anonymous namespace in serializer.cc. There's no way to make such functions `friends` in some other class. I promoted `DecodeResourceName` to be a member of the `Serializer` to gain access to `ResourcePath::FromStringView` and that caused some of the other functions that use `DecodeResourceName` to also need to be promoted to members also.

Reply:
Ah, I meant `static` in the sense "a `static` member function", not in the sense "a free function with static linkage". What I'm getting at is that a `static` function is more restricted than a `const` function (because it cannot access any class state at all), which in turn is more restricted than a non-`const` function, so if the change is purely about access control, it might make sense to  use the most restricted concept that works. Not a big deal, though -- just want to make my original comment clear.

Reply:
Ah! Thanks for clarifying.

I made these methods consistent with the other private methods in here like `EncodeNull` and other similar methods that don't need access to `this` but are nevertheless `const`. 
As far as I can tell, the only static members here are ones that are required publicly to be independent of any particular serializer instance.

If we want to revisit this more broadly we can address these additions then.